server/controllers/nlp.js

```diff
@@ -47,12 +208,76 @@ exports.handleInput = function (req, res) {
 
         let dataset = req.body.dataset;
 
+        let lowerCaseInput = req.body.input.toLowerCase();
+
+        var tokenizer = new natural.WordTokenizer();
+        let tokenizedInput = tokenizer.tokenize(lowerCaseInput);
+
+
+        let initState = {
+            tokens: tokenizedInput,
+            layer: 0,
+            currentToken: 0,
+            callback: state => {
+                let bestMatch = {
+                    function: 'plotPieChartOfColumn',
+                    parameters: []
+                };
+                var numberMatches;
+                var numberColumns;
+                var columnsArray;
+                var bestMatched = {
+                    numberMatches: 0,
+                    function: "",
+                    functionParameter: []
+                };
+                _.forEach(commands, function (command) {
+                    numberMatches = 0;
+                    numberColumns = 0;
+                    columnsArray = [];
+                    _.forEach(state.tokens, function (token) {
+                        if (token.type === "Column") {
+                            numberColumns++;
+                            columnsArray.push(token.value);
+                        } else if(token.type === "ChartType"){
+                            if(token.value === command.parameters.chartType){
+                                numberMatches++;
+                            }
+                        }
+                    });
+                    if(numberColumns === command.parameters.numberColumns){
+                        numberMatches++;
+                    }
+                    if(numberMatches > bestMatched.numberMatches){
+                        bestMatched.numberMatches = numberMatches;
+                        bestMatched.function = command.function;
+                        _.forEach(command.functionParameters, function(param, index){
+                            if(param === "Column"){
```

Comment:
what is the purpose of this ? it will always be true as command.functionParameters = "column"

Reply:
Well, it depends what we need to tansform the data. For now it just uses columns, but maybe later it needs also other input like if we should plot percent or total count.